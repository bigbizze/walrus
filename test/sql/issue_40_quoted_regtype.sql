/*
Tests that, regtypes that require quoting are handled without exception
*/

select 1 from pg_create_logical_replication_slot('realtime', 'wal2json', false);


create type "Color" as enum ('RED', 'YELLOW', 'GREEN');

create table public.notes(
    id int primary key,
    primary_color "Color"
);


create policy rls_color_is_red
on public.notes
to authenticated
using (primary_color = 'RED');

alter table public.notes enable row level security;


insert into realtime.subscription(subscription_id, entity, claims, filters)
select
    seed_uuid(1),
    'public.notes',
    jsonb_build_object(
        'role', 'authenticated',
        'sub', seed_uuid(2)::text
    ),
    array[('primary_color', 'eq', 'RED')::realtime.user_defined_filter];

insert into public.notes(id, primary_color)
values
    (1, 'RED'),   -- matches filter
    (2, 'GREEN'); -- does not match filter

select
    rec,
    is_rls_enabled,
    subscription_ids,
    errors
from
   walrus;


drop table public.notes;
select pg_drop_replication_slot('realtime');
truncate table realtime.subscription;
