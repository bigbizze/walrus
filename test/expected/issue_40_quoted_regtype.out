/*
Tests that, regtypes that require quoting are handled without exception
*/
select 1 from pg_create_logical_replication_slot('realtime', 'wal2json', false);
 ?column? 
----------
        1
(1 row)

create type "Color" as enum ('RED', 'YELLOW', 'GREEN');
create table public.notes(
    id int primary key,
    primary_color "Color"
);
insert into realtime.subscription(subscription_id, entity, claims, filters)
select
    seed_uuid(1),
    'public.notes',
    jsonb_build_object(
        'role', 'authenticated',
        'sub', seed_uuid(2)::text
    ),
    array[('primary_color', 'eq', 'RED')::realtime.user_defined_filter];
insert into public.notes(id, primary_color)
values
    (1, 'RED'),   -- matches filter
    (2, 'GREEN'); -- does not match filter
select
    rec,
    is_rls_enabled,
    subscription_ids,
    errors
from
   walrus;
                      rec                       | is_rls_enabled |            subscription_ids            | errors 
------------------------------------------------+----------------+----------------------------------------+--------
 {                                             +| f              | {f4539ebe-c779-5788-bbc1-2421ffaa8954} | {}
     "type": "INSERT",                         +|                |                                        | 
     "table": "notes",                         +|                |                                        | 
     "record": {                               +|                |                                        | 
         "id": 1,                              +|                |                                        | 
         "primary_color": "RED"                +|                |                                        | 
     },                                        +|                |                                        | 
     "schema": "public",                       +|                |                                        | 
     "columns": [                              +|                |                                        | 
         {                                     +|                |                                        | 
             "name": "id",                     +|                |                                        | 
             "type": "int4"                    +|                |                                        | 
         },                                    +|                |                                        | 
         {                                     +|                |                                        | 
             "name": "primary_color",          +|                |                                        | 
             "type": "Color"                   +|                |                                        | 
         }                                     +|                |                                        | 
     ],                                        +|                |                                        | 
     "commit_timestamp": "2000-01-01T00:01:01Z"+|                |                                        | 
 }                                              |                |                                        | 
 {                                             +| f              | {}                                     | {}
     "type": "INSERT",                         +|                |                                        | 
     "table": "notes",                         +|                |                                        | 
     "record": {                               +|                |                                        | 
         "id": 2,                              +|                |                                        | 
         "primary_color": "GREEN"              +|                |                                        | 
     },                                        +|                |                                        | 
     "schema": "public",                       +|                |                                        | 
     "columns": [                              +|                |                                        | 
         {                                     +|                |                                        | 
             "name": "id",                     +|                |                                        | 
             "type": "int4"                    +|                |                                        | 
         },                                    +|                |                                        | 
         {                                     +|                |                                        | 
             "name": "primary_color",          +|                |                                        | 
             "type": "Color"                   +|                |                                        | 
         }                                     +|                |                                        | 
     ],                                        +|                |                                        | 
     "commit_timestamp": "2000-01-01T00:01:01Z"+|                |                                        | 
 }                                              |                |                                        | 
(2 rows)

drop table public.notes;
select pg_drop_replication_slot('realtime');
 pg_drop_replication_slot 
--------------------------
 
(1 row)

truncate table realtime.subscription;
